pilot.go

```diff
@@ -160,6 +108,63 @@ func initAutoPilot(svr *server, cfg *autoPilotConfig) (*autopilot.Agent, error)
 		},
 		Graph:           autopilot.ChannelGraphFromDatabase(svr.chanDB.ChannelGraph()),
 		MaxPendingOpens: 10,
+		ConnectToPeer: func(target *btcec.PublicKey, addrs []net.Addr) error {
```

Comment:
Can have this return a bool to denote if we're already connected. That way we can avoid disconnecting a node that we already have an active connection to.